controllers/modals/editorDecision/form/EditorDecisionWithEmailForm.inc.php

```diff
@@ -33,7 +33,7 @@ function getSaveFormOperation() {
 
 	/**
 	 * Set the operation to save this form.
-	 * @param $saveFormOperation string
```

Comment:
@jonasraoni, what self-doc tool are you structuring this for? I haven't looked at the config file recently (`docs/dev/ojs2.doxygen`) but Doxygen prefers the original order.

Reply:
I've seen this syntax in recent code and I thought we were moving to the [standard proposal](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md). Let me know if I should switch back, I didn't check how bad it will be in doxygen.

Reply:
Yeah, Alec, I've been meaning to bring this up. The format recognized by PHPIntelephense and used by [Laravel](https://github.com/illuminate/database/blob/master/ConnectionInterface.php#L9-L15) is:

```
@param Type $variable
```

I think that's become the standard for php and that's what my code editor expects for syntax highlighting. I did a bit of looking for a way to change these in bulk with php-cs but haven't found a rule yet.

Reply:
Confirmed that PSR-5 is a good target. And good news -- while [some sources](https://gist.github.com/billcreswell/5066109#file-phpprojectdocumentationhints-L69) recommend using `@param $paramName type` with Doxygen, as we've often done in the past, I've just tested and Doxygen is fine with `@param type $paramName` (it appears Doxygen looks for the `$`). (And I'm not wedded at all to Doxygen if phpDocumentor is a better tool for our purposes.)

Unfortunately I can't find a definitive plan for how to reconcile docblock `@type` declarations and PHP strict typing -- maybe either of you know? -- so until we get something definitive let's follow PSR-5 and cautiously add strict typing to functions. (I say cautiously because our current self-doc has not been strictly enforced and we're likely to run into cases where nulls or strings vs. ints introduce complaints in strict typing that might not be expected. So for example please avoid broad additions of strict typing to stable branches in particular.)

Reply:
Reconciliation: It sounds a bit awkward to declare the type twice, but the `PSR-5` is using the word "MUST" when talking about types for the `@param` and `@return`... I think it should be more flexible, but that's it 😁

Strict typing: Yeah, enforcing types and changes to the visibility of properties/methods (I saw some pieces of code making use use of `_private` things, so a global search will be needed before touching it) is always risky without a compiler 👀

Adoption: In case @NateWr can't find something in phpcs to replace it automatically, I think it's safe to use a regexp, since the maximum we'll get is an invalid docblock (which might be flagged later by phpcs) 🤔

Reply:
I can take care of a `sed` script to swap these around, but I'd rather wait for a month or two until we're doing less cherry-picking between `master` and `stable-3_3_0`.